frameworks/kafka/tests/auth.py

```diff
@@ -170,6 +170,10 @@ def write_failed(output) -> bool:
             if "LEADER_NOT_AVAILABLE" in stderr:
                 LOG.warning("Write failed with LEADER_NOT_AVAILABLE")
                 return True
+            if not is_not_authorized(output):
+                LOG.warning("Write failed with AuthorizationException")
+                return False
+            raise Exception("Not yet captured error while sending")
```

Comment:
What about `"Unexpected error while sending"`?

Reply:
Does reading need similar changes?

Reply:
In the read method, all the conditional branches are handled.